src/parsers.rs

```diff
@@ -195,7 +196,7 @@ impl<'a> Parser<'a> {
                 tail[*tail_idx] = part as u16;
                 *tail_idx += 1;
 
-                if part == 0xffff && use_tail && *tail_idx == 1 {
+                if *tail_idx == 1 && tail[0] == 0xffff {
```

Comment:
Also need to make sure that `head_idx` is 0, otherwise `1::ffff:<ip>` is valid.

Reply:
per my reading of the spec, `0::ffff:<ip>` is valid though. i ended up checking that all of `head` is 0 in the tail match arm. tests added too.

Reply:
:+1: